compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -355,6 +357,21 @@ FnSymbol* build_accessor(AggregateType* ct, Symbol* field,
     }
   }
 
+  Symbol* baseSym = _this;
+  if (typeOrParam && chapelClass) {
+    // Strip the _owned/_shared wrapper, if applicable.
+    VarSymbol* borrowOfThis = new VarSymbol("borrowOfThis");
+    if (typeMethod) borrowOfThis->addFlag(FLAG_TYPE_VARIABLE);
+
+    fn->insertAtTail(new DefExpr(borrowOfThis));
+    // This cast "throws" when the receiver actual is nilable. OK because:
+    // - If this field is a runtime type, we will generate an error below.
+    // - Otherwise this method will be eliminated before lowerErrorHandling.
```

Comment:
If you don't expect it to ever `throw` in practice I think it'd be better to use `!` that halts

Reply:
`!` is a less favorable option because:
* the receiver can be a type, if this is an accessor  to a type, not an instance; given that we are depreciating that
* `!` returns `unmanaged` when the accessor is invoked on an unmanaged flavor - of either type or value; `unmanaged` is a DecoratedClass type and applying PRIM_GET_MEMBER to it does not work because returnInfoGetMember expects an AggregateType.

These potentially-throwing casts are inserted into accessor methods only for param and type fields, so they are not actually invoked at run time. (If they were, the tests I added would not compile.)